spec/vector_spec.rb

```diff
@@ -449,6 +449,97 @@
               index: @multi_index, name: :precise_assignment, dtype: dtype))
           end
         end
+        
+        context Daru::CategoricalIndex do
+          context "non-numerical index" do
+            let (:idx) { Daru::CategoricalIndex.new [:a, :b, :a, :a, :c] }
+            let (:dv)  { Daru::Vector.new 'a'..'e', index: idx1 }
+
+            context "single category" do
+              context "multiple instances" do
+                subject { dv }
+                dv[:a] = 'x'
+
+                it { is_expected.to be_a Daru::Vector }
```

Comment:
Is this check necessary for each spec? I don't think `dv` could change its class after `#[]=` even in case of hugest error possible, so, what exactly we a specifying here?.. :)